src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -364,6 +367,34 @@ private static void validateType(
     }
   }
 
+  /**
+   * Attach {@code sampleParamConfig} to the nodes in this tree. Also sets lineType to {@code
+   * InitCodeLineType.ReadFileInitLine} if specified in {@code sampleParamConfig}.
+   *
+   * @param parentFieldPath The full path of the parent object of {@code typeRef}. Set to an empty
+   *     string if {@code typeRef} is a top level proto object. We need to keep track of this
+   *     because the keys in {@code sampleParamConfigMap} are full paths while {@code key} is a
+   *     simple field name.
+   */
+  private void resolveSampleParamConfigs(
+      ImmutableMap<String, SampleParameterConfig> sampleParamConfigMap, String parentFieldPath) {
+    String fieldPath;
+    if (ROOT_KEY.equals(key)) {
+      fieldPath = "";
+    } else if (parentFieldPath.isEmpty()) {
+      fieldPath = key;
+    } else {
+      fieldPath = parentFieldPath + "." + key;
+    }
+    this.sampleParamConfig = sampleParamConfigMap.get(fieldPath);
+    if (sampleParamConfig != null && sampleParamConfig.readFromFile()) {
+      setLineType(InitCodeLineType.ReadFileInitLine);
```

Comment:
If the condition is false, do you need to `setLineType` to something else?

Reply:
It would be an no-op. The `lineType` is handled at different places, e.g. `simpleLineType` [here](https://github.com/googleapis/gapic-generator/blob/c12a7da29e0571a940664c4fa17080cf096b7dae/src/main/java/com/google/api/codegen/metacode/InitCodeNode.java#L308).